python/lsst/ip/diffim/subtractImages.py

```diff
@@ -581,6 +576,42 @@ def finalize(self, template, science, difference, kernel,
             correctedExposure = difference
         return correctedExposure
 
+    def updateMasks(self, difference, science, template):
+        # Update the mask planes on images for finalizing.
+
+        bbox = science.getBBox()
+        mask = difference.mask
+        mask &= ~(mask.getPlaneBitMask("DETECTED") | mask.getPlaneBitMask("DETECTED_NEGATIVE"))
+
+        if "FAKE" in science.mask.getMaskPlaneDict().keys():
+            # propagate the mask plane related to Fake source injection
+            # NOTE: the fake source injection sets FAKE plane, but it should be INJECTED
+            # NOTE: This can be removed in DM-40796
+
+            self.log.info("Adding injected mask planes")
+            mask.addMaskPlane("INJECTED")
+            diffInjectedBitMask = mask.getPlaneBitMask("INJECTED")
+
+            mask.addMaskPlane("INJECTED_TEMPLATE")
+            diffInjectedTemplateBitMask = mask.getPlaneBitMask("INJECTED_TEMPLATE")
+
+            scienceFakeBitMask = science.mask.getPlaneBitMask('FAKE')
+            templateFakeBitMask = template[bbox].mask.getPlaneBitMask('FAKE')
+
+            injScienceMaskArray = ((science.mask.array & scienceFakeBitMask) > 0)
+            injScienceMaskArray *= diffInjectedBitMask
```

Comment:
In testing, I get the following error from this line (and below): 
```numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'multiply' output from dtype('int64') to dtype('bool') with casting rule 'same_kind'```
I suggest removing the line `injScienceMaskArray *= diffInjectedBitMask` and using `mask.array[injScienceMaskArray] |= diffInjectedBitMask` below. You might want to change the variable name of `injScienceMaskArray` to reflect the new usage.